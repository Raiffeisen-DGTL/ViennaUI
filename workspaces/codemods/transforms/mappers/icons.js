export const iconsMapper = {
    ArrowRight: 'Forward',
    Exclamation: 'WarningRing',
    Down: 'SelectOpenDown',
    ArrowLeftRing: 'BackRing',
    Close: 'Close',
    Invisible: 'EyeClosed',
    ArrowUp: 'Up',
    CheckSign: 'SealCheck',
    CheckMark: 'Checkmark',
    CheckMarkRing: 'CheckmarkRing',
    Infinity: 'Infinity',
    Attention: 'WarningTr',
    Exit: 'Exit',
    ArrowsDiagonal: 'ArrowsDiag',
    Burger: 'BurgerHor',
    Favorite: 'FavEmpty',
    ArrowDownRing: 'DownRing',
    ArrowDown: 'Down',
    Buttons: 'KeyboardSettings',
    ArrowsLeftRightBlackRgb: 'ArrowsLeftRight',
    Filter: 'Filter',
    ArrowsUpDownBlackRgb: 'ArrowsTopDown',
    Download: 'Download',
    CircleCheckedDashed: 'CheckmarkPart',
    InfoRing: 'InfoRing',
    CloseRing: 'ClearRing',
    Left: 'GoLeft',
    Compass: 'Safari',
    LeftArrow: 'Back',
    CompassArrow: 'Location',
    ArrowRightRing: 'ForwardRing',
    Coordinates1: 'Coordinates',
    ArrowUpRing: 'UpRing',
    Copy: 'Copy',
    DragAndDrop: 'Drag1',
    Draggable: 'Drag',
    Enter: 'Enter',
    Pin2: 'TerminalOnMap',
    Stop: 'Cancel',
    Plus: 'Add',
    SubmitRefillLink: 'Share',
    OtherRing: 'CurrentRing',
    Mirror: 'Mirror1',
    Queue: 'InQueue',
    Up: 'SelectHide',
    PlusRing: 'AddRing',
    Trash: 'Trash',
    Lock: 'Locked',
    Other: 'MoreHor',
    Settings1: 'Settings',
    Sorting: 'Sorting',
    LogoArrow: 'Pin1',
    More: 'MoreVert',
    Refresh3: 'RefreshRepeat',
    ViewCards: 'Grid',
    Refresh2: 'RefreshRepeat',
    ViewList: 'BurgerHor',
    Refresh: 'Retry1',
    QuestionFill: 'HintFilled',
    Rewind: 'RewindLeft',
    InfoFill: 'InfoFilled',
    RewindRight2: 'CollapseRight',
    RepeatWithErrors: 'RetryWithErrors',
    RewindLeft2: 'CollapseLeft',
    Unlock: 'Unlocked',
    RewindRight: 'RewindRight',
    Question: 'HintRing',
    Right: 'GoRight',
    Save: 'Save1',
    Visible: 'EyeOpened',
    InQueue: 'ListOpen',
    InQueueBack: 'ListClose',
    Unarchive: 'Unarchive',
    Diagram: 'ChartPie',
    Schedule: 'ChartBar2',
    Schedule3: 'ChartLineGrow1',
    ScheduleGrowth: 'ChartBarGrow2',
    Scheme: 'Scheme1',
    Scheme2: 'Scheme2',
    Accounts: 'Docs',
    CheckMagnifier: 'InvoiceSearch',
    Task: 'TaskDone',
    ByReceiptNumber: 'DocNmb',
    ClipBlack: 'Attach',
    DocumentSignAndSend: 'DocSignAndSend',
    DocumentSignVisa: 'DocSignVisa',
    AccountError: 'DocError',
    Document2: 'Clipboard',
    Document3: 'Task',
    Document3Arrow: 'TaskSend',
    DocumentFavorites: 'DocFav',
    Envelope: 'Mail',
    Excel: 'Xls',
    Incoming: 'Mail',
    BetweenYourAccounts: 'AccOperations',
    NewDocument: 'DocNew',
    BriefcaseInterest: 'BriefcaseInterest',
    OfficeSupplies: 'OfficeSupplies',
    Organizations: 'Case',
    Passport: 'Passport',
    Pen: 'Pen',
    Pencil: 'Edit',
    SignChecked: 'DocSignChecked',
    SaveAsTemplate: 'TemplateSave',
    CheckSeal: 'InvoiceSeal',
    Template: 'Template',
    Check: 'Invoice2',
    CurrentAccount: 'DocVerified',
    Seal: 'SealStamp',
    SignPartlySigned: 'PaymentPartlySigned',
    ToYourAccount: 'DocRecieve',
    Check2: 'Invoice1',
    DictionaryAdd: 'AddToDic',
    SignRemove: 'DocSignRemove',
    Schedule2: 'ChartBar1',
    VerifiedTemplate: 'TemplateSaved',
    CheckInfo: 'InvoiceInfo',
    Document: 'Document',
    SignSigned: 'PaymentSigned2',
    DocumentCopy: 'Copy',
    TargetAccount: 'AccTarget',
    DocumentEdit: 'DocEdit',
    DocumentHistory: 'DocHistory',
    SavingsAccount: 'Savings',
    DocumentSeal: 'DocSeal',
    DocumentReissue: 'DocReissue',
    EditBlackRgb: 'Edit',
    DocumentSend: 'DocSend',
    CashWithdrawal: 'AtmCash1',
    Terminal4: 'Terminal',
    Transfer: 'CardToOtherBank',
    Something: 'CardRaif',
    Card: 'CardLock',
    TransferToCard: 'C2c',
    Terminal: 'Acquiring1',
    Cards: 'Cards',
    CardLock: 'CardLock',
    CardHeart: 'CardHeart',
    Open: 'CardAddNew',
    Terminal2: 'Acquiring2',
    Mastercard: 'CardMaster',
    Terminal3: 'Pos',
    Atm: 'AtmCash1',
    Atmcard: 'AtmCards1',
    Bridge: 'Bridge',
    ByCardNumber: 'CardNmb',
    Card1: 'Card',
    CardPinChange: 'CardChangePin',
    DepositFunds: 'DepositOperations',
    HandMoney: 'HandCash1',
    Contribution: 'MoneyBag',
    BundleOfMoney: 'BundleOfMoney',
    Pif: 'Pif',
    RubleTurnover: 'RoubleReturn',
    Calculator: 'Calculator',
    CurrencyEur: 'Euro',
    Euro: 'Euro',
    RubleRadial: 'RoubleRing',
    Exchange: 'Exchange2',
    ToPay: 'ToPay',
    HandMoney2: 'HandCash2',
    DebtFcpp: 'DebtFcpp',
    MoneyBag: 'MoneyBag',
    Ruble: 'Rouble',
    Dollar: 'DollarUs',
    Pay: 'Wallet2',
    RubleArrow: 'RoubleSend',
    CashTransactions: 'CashTransactions',
    PaymentOnReceipt1: 'DocRouble',
    CirculationOfMoney: 'RoubleCashflow',
    GearWheelRuble: 'RoubleGear',
    PlusMinus: 'PlusMinus',
    Credit: 'CreditCash',
    HandCard: 'CardGet',
    Portfolio: 'Case',
    CurrencyCalculator: 'CalculatorCurrency2',
    Purse2: 'Wallet1',
    Interest: 'Percent',
    Pear: 'Pear',
    Coffee: 'Coffee',
    Man2: 'Man',
    ServicesAndService: 'PersonService',
    People: 'People2',
    Furniture: 'Furniture',
    Cocktail: 'Cocktail',
    Cake: 'Cake',
    Grandfather: 'Grandfather',
    SmileNoMouth: 'FaceNoMouth',
    Candles: 'FireHouse',
    Flower: 'Flower',
    Consultation: 'Consultation',
    Tray: 'FoodService',
    LampHome: 'Lamp',
    SmileSad: 'FaceSad',
    BabyCare: 'People1',
    Stroller: 'Stroller',
    IceCream: 'IceCream',
    Champagne: 'GlassesStemware',
    Ambulance: 'Ambulance',
    Chair: 'Chair',
    Key: 'DocSign',
    TransferToAPrivatePerson: 'ToPerson',
    Girl: 'Girl',
    Window: 'Window',
    Cafe: 'Cafe',
    Man: 'Man',
    Smile: 'FaceSmile',
    Woman: 'Women',
    Grandmother: 'Grandmother',
    ManButterfly: 'ManButterfly',
    CupSport: 'Rating',
    Heart: 'Heart',
    ManRing: 'ManRing',
    SmileNormal: 'FaceNeutral',
    Candy: 'Candy',
    Cupboard: 'Cupboard',
    HeartBroken: 'HeartBroken',
    Men: 'People1',
    Cap: 'Cook',
    Fireplace: 'Fireplace',
    Monitor: 'Monitor',
    PoliceCap: 'PoliceCap1',
    Shield2: 'Shield',
    Policeman: 'ArmyCap',
    Libra: 'Libra',
    Deposit: 'Safe',
    Bell: 'Notification',
    Flasher: 'Flasher',
    PoliceCar: 'PoliceCar1',
    Fingerprint: 'Prints2',
    Court: 'Court',
    Shield: 'Insurance',
    ShieldCheck: 'ShieldCheck',
    Design: 'Wacom',
    PostServices: 'Frame',
    Theater: 'TheaterOut',
    Violin: 'Violin',
    Bust: 'Bust',
    Literature: 'Literature',
    Movie: 'Movie',
    Art: 'Music1',
    Picture: 'Picture1',
    Brush: 'Brush',
    Barrel: 'Barrel',
    Box: 'Archive',
    Canister: 'Canister',
    ConcreteMixer: 'ConcreteMixer',
    Equipment: 'Equipment',
    Excavator: 'Excavator',
    Factory: 'Factory',
    Gas: 'Gas',
    Gears: 'Gears',
    Hammers: 'Repair1',
    Lamp: 'LightBulbLamp',
    Oil: 'Oil',
    Pipes: 'Pipes',
    Screw: 'Screw',
    Tower: 'Tower',
    Tractor: 'Tractor',
    Truck: 'Truck',
    Water: 'Water',
    WateringCan: 'WateringCan',
    Wheelbarrow: 'Wheelbarrow',
    Satellite: 'Satellite3',
    Chat: 'Chat2',
    Print: 'Printer',
    Keyboard: 'Keyboard',
    BabbleQuestion: 'ChatAsk',
    SendToMail: 'MailIn',
    Support: 'Headphones1',
    Sms: 'Chat1',
    Sound: 'Sound',
    PhoneInfo: 'MobileInfo',
    QuestionLogo: 'ChatAskRaif',
    SatelliteDish: 'Satellite2',
    Entertainment: 'Console',
    Laptop: 'Laptop',
    CommunicationServices: 'MobileConnecting',
    Headphones: 'Headphones2',
    Bobbin: 'Cinema',
    Camera: 'CameraPhoto',
    Camera1: 'CameraVideo ',
    Babble: 'ChatGroup',
    BabbleLogo: 'ChatRaif',
    Mail: 'MailOut',
    Phone: 'Phone',
    PhoneHome: 'Telephone',
    Microphone: 'Microphone',
    AlcoholTobacco: 'AlcoholAndTobacco',
    Packages: 'OtherGoods',
    PriceTag: 'TagPrice3',
    Nameplate: 'NamePlate',
    BabyClothes: 'ClothesBaby',
    Basket2: 'Grocery2',
    Cashbox: 'Cashbox',
    CarDealers: 'CarDealers',
    Purchases: 'Purchases',
    RentACar: 'CarRent',
    QrCode: 'CodeQr',
    CodeRing: 'CodeBarRing',
    Code: 'CodeBar 2',
    Kids: 'Kids',
    HomeAndAppliances: 'HomeEquipment',
    Footwear: 'Footwear2',
    Beauty: 'Mirror1',
    Clothes: 'Clothes2',
    ClothesShoesAccessories: 'Clothes1',
    Cosmetics: 'Cosmetics2',
    GasStation: 'GasStation',
    HealthAndBeauty: 'Apple',
    Offer: 'Offer',
    PriceTagRuble: 'TagPrice1',
    Products: 'Basket',
    Treatment: 'Treatment2',
    Pharmacies: 'Pharmacy',
    Moon: 'Moon',
    Rain: 'Rain',
    PetSupplies: 'Pet',
    Garden: 'Garden',
    Dolphin: 'Dolphin',
    Cloud: 'Bear',
    Snow: 'Snow',
    Tree: 'Tree',
    Fish: 'Fish',
    Leaf: 'Leaf',
    Sun: 'Sun',
    SunCloud: 'SunCloud',
    Bear: 'Bear1',
    Umbrella2: 'UmbrellaClosed',
    Cow: 'Bull',
    Animal: 'Cat',
    Umbrella: 'UmbrellaOpened',
    Whale: 'Whale',
    Ufo: 'Ufo',
    Electronics: 'Monitor',
    DutyFree: 'CreditConsumer',
    Timer: 'Timer',
    Balloon: 'Balloon',
    Backpack2: 'Suitcase',
    Cup: 'Cup',
    Tent: 'Tent',
    TennisBall: 'TennisBall',
    Skates: 'Skates',
    Fishing: 'Fishing',
    Chess: 'Chess',
    Archer: 'Archer',
    PalmBlack: 'Palm',
    Backpack: 'Backpack',
    Barbell: 'Barbell',
    Bowling: 'Bowling',
    BoxSport: 'Boxing',
    ChaiseLounge: 'Sunshade',
    Darts: 'Darts',
    Drone: 'Drone',
    Flag: 'Flag',
    Goals: 'Goals',
    Golf: 'Golf',
    Leisure: 'Tennis',
    LeisureAndTravel: 'Paris',
    Map: 'MapSearch',
    Medal: 'Medal',
    Mountains: 'Mountains',
    Parachute: 'Parachute',
    PingPong: 'PingPong',
    Robot: 'Robot',
    Skiing: 'Skiing',
    Sport: 'Basketball',
    Tourism: 'Navigation',
    Toys: 'Bear',
    TrainingApparatus: 'Gym',
    Volleyball: 'Volleyball',
    Wave2: 'Boat',
    Wave: 'Wave',
    BankHome: 'BankAndHome',
    HouseWap: 'HouseWap',
    BankIntroduction: 'BranchIn',
    Building: 'Company',
    Bank: 'Branch',
    Castle: 'Castle',
    Crane: 'Crane1',
    Crane2: 'Crane2',
    Fence: 'Fence',
    Heating: 'Heating2',
    Home: 'Home',
    Hotels: 'Hotels',
    Houses: 'City',
    Houses2: 'Houses',
    Repairs: 'Repair2',
    ToBudget: 'ToBudget',
    GeneralEducation: 'Book',
    ArtAndLiterature: 'ArtAndLiterature',
    Binoculars: 'Binoculars1',
    Binoculars2: 'Binoculars2',
    BooksShelf: 'BooksShelf',
    Courses: 'Courses',
    Education: 'Education',
    Glasses: 'GlassesEye',
    Globe: 'Globe2',
    Magnifier: 'Search',
    Microscope: 'Microscope',
    ProfessionalEducation: 'Repair1',
    WorldCopy: 'World',
    Metro: 'Metro',
    Gbdd: 'Fines',
    Cruises: 'Ship',
    Auto: 'AutotransportFront',
    Rocket: 'Rocket',
    Taxi2: 'Taxi',
    Taxi: 'Taxi',
    Speedometer: 'Speedometer',
    MetroTrain: 'Railway2',
    AutoParts: 'CarParts',
    Moto: 'Moto1',
    TrolleyBus: 'TrolleyBus',
    AirTransport: 'Airplane',
    AutoService: 'Wrench',
    Bicycle: 'Moto2',
    Helicopter: 'Helicopter',
    MotorTransport: 'Bus',
    Railway: 'Railway',
    Calendar: 'Calendar',
    Clock: 'History',
    OperationsHistory: 'Retry2',
    Expectation: 'InProgress',
    Accessories: 'Accessories',
    AdEvent: 'CalendarAdd',
    Telegram: 'Telegram',
    Whatsapp: 'Whatsapp',
    Viber: 'PhoneActive',
    Clip: 'Attach',
};
