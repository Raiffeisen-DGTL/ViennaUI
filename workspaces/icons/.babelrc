{
  "presets": ["@babel/env", "@babel/typescript", "@babel/react"],
  "env": {
    "esm": {
      "presets": [["@babel/env", { "modules": false }]]
    }
  }
}
