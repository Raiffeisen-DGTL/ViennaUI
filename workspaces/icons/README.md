# Библиотека иконок

###### Raiffeisenbank Design System

Коллекция иконок для использования в интерфейсах. Иконки представлены в виде React компонентов и оригинальных векторных (`*.svg`) изображений. Изображения напрямую можно использовать из пакета.
