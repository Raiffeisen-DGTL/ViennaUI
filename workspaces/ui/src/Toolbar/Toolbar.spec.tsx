import React from 'react';
import { Screw } from 'vienna.icons';
import { Toolbar } from './index';
import { DropList } from '..';

it('Toolbar', () => {
    const toolbar = snapshot.shallow(
        <Toolbar>
            <Toolbar.Operation label={'Option 1'} />
            <Toolbar.Operation label={'Option 2'} />
            <Toolbar.Operation label={'Option 3'} />
            <Toolbar.Operation label={'Option 4'} />
        </Toolbar>
    );

    expect(toolbar.find(Toolbar.Operation)).toMatchSnapshot();
});

it('Toolbar /w dark', () => {
    const toolbar = snapshot.shallow(
        <Toolbar design='dark'>
            <Toolbar.Operation label={'Option 1'} />
            <Toolbar.Operation label={'Option 2'} />
            <Toolbar.Operation label={'Option 3'} />
            <Toolbar.Operation label={'Option 4'} />
        </Toolbar>
    );

    expect(toolbar.find(Toolbar.Operation)).toMatchSnapshot();
});

test('Toolbar w/ any props', () => {
    const props = {
        className: 'className',
        id: 'id',
        title: 'title',
    };

    const component = (
        <Toolbar {...props}>
            <Toolbar.Operation label={'Option 1'} />
            <Toolbar.Operation label={'Option 2'} />
            <Toolbar.Operation label={'Option 3'} />
            <Toolbar.Operation label={'Option 4'} />
        </Toolbar>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});

test('Toolbar w/ operation', () => {
    const component = <Toolbar.Operation label={'Action 5'} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('Toolbar w/ operation with icon', () => {
    const component = <Toolbar.Operation icon={<Screw />} label={'Action 5'} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('Toolbar w/ operation onClick()', () => {
    const fn = jest.fn();
    const component = <Toolbar.Operation id={'id'} name={'name'} icon={<Screw />} label={'Action 5'} onClick={fn} />;

    const snap = snapshot.shallow(component);

    snap.simulate('click', {});
    expect(fn).toHaveBeenCalledWith({}, { id: 'id', name: 'name' });
});

test('Toolbar w/ operation hover', () => {
    const component = (
        <Toolbar.Operation label={'Action 3'}>
            <Toolbar.Operation label={'Option 1'} />
            <Toolbar.Operation label={'Option 2'} />
            <Toolbar.Operation label={'Option 3'} />
        </Toolbar.Operation>
    );

    const snap = snapshot.shallow(component);

    let droplist = snap.find(DropList).get(0);
    expect(droplist).toBeUndefined();

    snap.simulate('mouseenter', {});
    snap.update();
    droplist = snap.find(DropList).get(0);
    expect(droplist).toBeDefined();

    snap.simulate('mouseleave', {});
    snap.update();
    droplist = snap.find(DropList).get(0);
    expect(droplist).toBeUndefined();
});
