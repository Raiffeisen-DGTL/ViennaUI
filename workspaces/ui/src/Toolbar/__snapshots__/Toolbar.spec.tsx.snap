// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar /w dark 1`] = `
Array [
  <Toolbar.Operation
    design="dark"
    key="0"
    label="Option 1"
  />,
  <Toolbar.Operation
    design="dark"
    key="1"
    label="Option 2"
  />,
  <Toolbar.Operation
    design="dark"
    key="2"
    label="Option 3"
  />,
  <Toolbar.Operation
    design="dark"
    key="3"
    label="Option 4"
  />,
  <Toolbar.Operation
    design="dark"
    hideElipsis={true}
    icon={
      <MoreHor
        color="currentColor"
        size="m"
      />
    }
    label="More operation"
  >
    <Toolbar.Operation
      asSubOperation={true}
      key="0"
      label="Option 1"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="1"
      label="Option 2"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="2"
      label="Option 3"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="3"
      label="Option 4"
    />
  </Toolbar.Operation>,
  <Toolbar.Operation
    asSubOperation={true}
    key="0"
    label="Option 1"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="1"
    label="Option 2"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="2"
    label="Option 3"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="3"
    label="Option 4"
  />,
]
`;

exports[`Toolbar 1`] = `
Array [
  <Toolbar.Operation
    design="light"
    key="0"
    label="Option 1"
  />,
  <Toolbar.Operation
    design="light"
    key="1"
    label="Option 2"
  />,
  <Toolbar.Operation
    design="light"
    key="2"
    label="Option 3"
  />,
  <Toolbar.Operation
    design="light"
    key="3"
    label="Option 4"
  />,
  <Toolbar.Operation
    design="light"
    hideElipsis={true}
    icon={
      <MoreHor
        color="currentColor"
        size="m"
      />
    }
    label="More operation"
  >
    <Toolbar.Operation
      asSubOperation={true}
      key="0"
      label="Option 1"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="1"
      label="Option 2"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="2"
      label="Option 3"
    />
    <Toolbar.Operation
      asSubOperation={true}
      key="3"
      label="Option 4"
    />
  </Toolbar.Operation>,
  <Toolbar.Operation
    asSubOperation={true}
    key="0"
    label="Option 1"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="1"
    label="Option 2"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="2"
    label="Option 3"
  />,
  <Toolbar.Operation
    asSubOperation={true}
    key="3"
    label="Option 4"
  />,
]
`;

exports[`Toolbar w/ any props 1`] = `
.c0 {
  outline: none;
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 62px;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  padding-left: 32px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-width: auto;
}

.c1:first-child {
  padding-left: 0px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  outline: 0;
  position: relative;
  border-radius: 8px;
  border-color: transparent;
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  font-size: 14px;
  line-height: 22px;
  padding: 0 20px;
  height: 36px;
  min-width: 36px;
  color: #2B2D33;
  background-color: transparent;
  padding: 0;
}

.c2:active {
  top: 1px;
}

.c2:hover {
  color: #00809C;
}

.c3 {
  display: inline-block;
  vertical-align: top;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3:not(:last-of-type) {
  margin-right: 8px;
}

.c3 > svg {
  display: block;
}

<div
  class="c0 className"
  id="id"
  title="title"
>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 1
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 2
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 3
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 4
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M4 10a2 2 0 110 4 2 2 0 010-4zm16 0a2 2 0 110 4 2 2 0 010-4zm-8 0a2 2 0 110 4 2 2 0 010-4z"
          />
        </svg>
      </span>
      <span
        class="c3"
      >
        More operation
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
</div>
`;

exports[`Toolbar w/ operation 1`] = `
.c0 {
  padding-left: 32px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-width: auto;
}

.c0:first-child {
  padding-left: 0px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  outline: 0;
  position: relative;
  border-radius: 8px;
  border-color: transparent;
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  font-size: 14px;
  line-height: 22px;
  padding: 0 20px;
  height: 36px;
  min-width: 36px;
  color: #2B2D33;
  background-color: transparent;
  padding: 0;
}

.c1:active {
  top: 1px;
}

.c1:hover {
  color: #00809C;
}

.c2 {
  display: inline-block;
  vertical-align: top;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2:not(:last-of-type) {
  margin-right: 8px;
}

.c2 > svg {
  display: block;
}

<div
  class="c0"
>
  <button
    class="c1"
    type="button"
  >
    <span
      class="c2"
    />
    <span
      class="c2"
    >
      Action 5
    </span>
    <span
      class="c2"
    />
  </button>
</div>
`;

exports[`Toolbar w/ operation with icon 1`] = `
.c0 {
  padding-left: 32px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-width: auto;
}

.c0:first-child {
  padding-left: 0px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  outline: 0;
  position: relative;
  border-radius: 8px;
  border-color: transparent;
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  font-size: 14px;
  line-height: 22px;
  padding: 0 20px;
  height: 36px;
  min-width: 36px;
  color: #2B2D33;
  background-color: transparent;
  padding: 0;
}

.c1:active {
  top: 1px;
}

.c1:hover {
  color: #00809C;
}

.c2 {
  display: inline-block;
  vertical-align: top;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2:not(:last-of-type) {
  margin-right: 8px;
}

.c2 > svg {
  display: block;
}

<div
  class="c0"
>
  <button
    class="c1"
    type="button"
  >
    <span
      class="c2"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          clip-rule="evenodd"
          d="M11.976 4.002a.437.437 0 00-.171.084c-.196.14-.46.449-.733 1.006-.266.544-.506 1.258-.693 2.12a16.89 16.89 0 00-.324 2.292A3.982 3.982 0 0112 9c.783 0 1.515.226 2.133.616.087.038.175.078.262.119a15.312 15.312 0 012.965 1.862 16.5 16.5 0 012.552 2.505c.719.887 1.283 1.805 1.616 2.681.326.857.465 1.768.188 2.575-.302.881-1.02 1.41-1.891 1.589-.82.169-1.766.035-2.712-.275-.963-.316-1.982-.83-2.973-1.485l1.102-1.669c.876.579 1.734 1.005 2.493 1.253.774.253 1.34.288 1.687.217.296-.06.366-.174.402-.28.062-.18.08-.57-.165-1.214-.238-.624-.675-1.36-1.301-2.133a14.508 14.508 0 00-2.359-2.29 3.99 3.99 0 01-1.177 2.764c-.165.22-.342.44-.531.66a16.166 16.166 0 01-2.618 2.405 15.764 15.764 0 01-3.049 1.78c-1.016.441-2.02.72-2.92.768-.882.047-1.78-.125-2.425-.723-.68-.632-.848-1.518-.716-2.378.13-.84.548-1.73 1.136-2.587.595-.869 1.383-1.735 2.288-2.53l1.32 1.502c-.803.706-1.473 1.45-1.958 2.158-.492.718-.742 1.324-.81 1.762-.064.42.048.558.1.607.088.082.352.224.958.191.59-.031 1.355-.224 2.232-.605.87-.377 1.784-.91 2.655-1.553.182-.135.36-.273.534-.413a4.002 4.002 0 01-2.974-4.476c-.122-1.911.008-3.894.38-5.615.212-.975.497-1.85.851-2.574.347-.71.794-1.346 1.367-1.755.613-.438 1.346-.592 2.08-.334.677.237 1.209.775 1.616 1.402.814 1.253 1.342 3.183 1.555 5.302l-1.99.2c-.2-1.979-.678-3.544-1.242-4.412-.281-.434-.504-.57-.6-.605a.155.155 0 00-.085-.01zM12 11a2 2 0 100 4.002A2 2 0 0012 11z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <span
      class="c2"
    >
      Action 5
    </span>
    <span
      class="c2"
    />
  </button>
</div>
`;
